### DISCUSSION TITLE
How to add a few strings for like "Select all"?

### DISCUSSION THREAD

**Discussion by Teut2711** [codingjoe/django-select2#33] [Q&A] [answered]
**Goal**
To get an entry "Select all" in the search.

**Problem**
I have around 20 million rows in table. The Admin wants to send mail to some specific or to all users. Now he may type a phrase to search for a user or he may want to send the mail to all users. If he wants to send mail to all the users it would just be sufficient for him to select "Select all" from the field.
 
**Code Snippet**
Please provide a code snippet of your problem.
```


class UsersWidget(s2forms.ModelSelect2MultipleWidget):
    search_fields = [
        "email__istartswith",
        "PAN_ID__istartswith",
        "full_name__istartswith",
    ]
    empty_label = "Select all"

    def label_from_instance(self, obj):
        return obj.email

    def filter_queryset(self, request, term, queryset=None, **dependent_fields):

        return super.filter_queryset(request, term, queryset, **dependent_fields)


class MailForm(forms.Form):
    to = forms.ModelMultipleChoiceField(
        label="To",
        queryset=User.objects.all(),
        widget=UsersWidget,
    )

    subject = forms.CharField(max_length=78)
    content = QuillFormField()
    fileupload = forms.FileField()

```
![image](https://user-images.githubusercontent.com/40588378/107181034-09175380-6a00-11eb-9171-f9a21a9fab83.png)

A badge of this kind with text as "Select all"  is required.

![image](https://user-images.githubusercontent.com/40588378/107182954-c5bee400-6a03-11eb-81a5-3ee59732353f.png)

**Comment by codingjoe** [comment, marked as answer]
Hi @XtremeGood, thanks for reaching out. IMHO, you will have an easier time, if you don't try to overload a select box for that. I would suggest you add a separate UI element, like a checkbox, which disables the select. I believe this to be not only technically simpler but also to prove to be the better user experience, since it is more explicit. I hope this helps you, even if I didn't really answer your questions ;) Best, Joe

**Reply by jrbosch** [reply]
Hi @codingjoe and @XtremeGood I’m in a similar situation because I’m using Select2MultipleWidget on a Django ModelForm related to a ManyToManyField model field that query for many elements. In my case if I use a separate UI element like a checkbox which disables the form field, this will make the POST not contain the list of elements and will not save the correct association of the models in the database (that is what I need). Maybe adding some extra code on backend could be an alternative but I don't think it is the easiest and most elegant way. Other thing is that some time I need to select all the element and after that just unselect a few of them, for this, a separate UI I think is unuseful. I still thinking that have a Select All option will be useful, but anyway I’m not a great web developer and maybe I’m missing something that could solve my needs, or I don’t know if this is possible right now using the Select2MultipleWidget constructor. I would appreciate any suggestions from you. Best Regards.

**Comment by Teut2711** [comment]
Thanks @codingjoe , that seems to be a better idea.
